did/web.go

```diff
@@ -0,0 +1,146 @@
+package did
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+)
+
+// did:web method specification
+// https://w3c-ccg.github.io/did-method-web/
+// DID Web create and resolve mothods are implemented in this package
+// but NOT the update and deactivate methods
+// please refer to web_test.go for example and test cases
+type DIDWeb string
+
+const (
+	DID_WEB_WELL_KNOWN_URL_PATH = ".well-known/"
+	DID_WEB_DID_DOC_NAME        = "did.json"
+)
+
+// CreateDoc constructs a did:web DIDDocument from a specific key type and its corresponding public key
+// This method does not attempt to validate that the provided public key is of the specified key type
+// The returned DIDDoucment is expected further turned into a JSON file named did.json
+// and stored under the expected path of the target web domain
+// specification: https://w3c-ccg.github.io/did-method-web/#create-register
+func (did DIDWeb) CreateDoc(kt crypto.KeyType, publicKey []byte) (*DIDDocument, error) {
+	//create DIDKey with publicKey
+	didKey, err := CreateDIDKey(kt, publicKey)
+	if err != nil {
+		return nil, err
+	}
+	//create DIDDocument
+	didDoc, err := didKey.Expand()
+	if err != nil {
+		return nil, err
+	}
+	didWebStr := string(did)
+	didDoc.ID = didWebStr
+	verMethodID := didWebStr + "#owner"
+	didDoc.VerificationMethod[0].ID = verMethodID
```

Comment:
I'm not certain how I feel about relying so heavily on the did key methods. They have a separate API, and could change. Changes in the did:key API **should not** break the `did:web` methods.

What do you think about re-implementing some of the DID creation logic, specific to this DID method?

Reply:
Good call. Will do.